docs/reference/async-processing.mdx

```diff
@@ -0,0 +1,116 @@
+
+### 3. Ensure that you implement the Processor#stores method
+
+Note that the async processing framework requires a stateful processor to connect its StateStores by implementing the `ProcessorSupplier#stores` method. Please verify that your processor supplier implements this method before passing it into the async wrapper, or switch to this method of connecting state stores if it does not.
+You will know if this is not set up correctly by an immediate `IllegalStateException` upon startup.
+
+## Tuning the AsyncProcessor
+
+Several new configs have been introduced as part of the async processing feature, but the framework also interacts with several existing Streams and client configs. See the advice below for our thoughts on the best possible setup to make the most of async processing for your application.
+
+### commit.interval.ms
+
+Although we always advise increasing the commit interval for EOS applications, it is especially to do so when enabling the async processing framework if you want to maximize its potential. If your application has only a few subtopologies (one or two), you can probably bump up the `commit.interval.ms` to 30s (which is the default for at-least-once-semantics). If your application has a larger number of subtopologies, the end-to-end latency will accrue across them, so it may be good to divide the 30s recommendation by a factor of the subtopology count. 
+
+:::note
+ When increasing the commit interval, you should always increase the `transaction.timeout.ms` producer config accordingly. We recommend simply setting this to 15s greater than the value of your chosen commit interval.
+ ::: 
+
+### num.stream.threads
+
+While an async processor will farm most of the processing work out to its async thread pool, the StreamThread still plays a vital role in moving records through the subtopology and issuing writes. It’s important that you still configure the node with an appropriate number of StreamThreads, most likely the same number that you would have used before enabling async processing. There will be more threads created in total, but keep in mind that the async threads will typically be spending most of their time waiting on the long RPC calls and not soaking up cpu cycles. Our advice: don’t touch the `num.stream.threads` config, except to make sure it’s set appropriately to begin with!
+
+### responsive.async.thread.pool.size
+
+One of the most important of the new configs is of course the `responsive.async.thread.pool.size`. Note that this value represents the number of async threads *per StreamThread*, so the actual number of async threads that will be created is equal to the `responsive.async.thread.pool.size` x `num.stream.threads`.  As always, the best way to choose a value for something like this is to start with an educated guess and then from there, experiment with your specific setup to find what works best for you. 
```

Comment:
obviously out of scope for this PR, but why didn't we choose to make the `async.thread.pool.size` a static number? seems like having one config depending on another config is quite confusing (@ableegoldman )

Reply:
I guess I personally found it easier to reason about in terms of "number of threads per StreamThread"

Also, that way it's always exact, vs if we make it a static number (I assume you mean like a config for the actual total number of async threads?) then the actual number of threads may not match if its not an exact multiple of the `num.stream.threads`

Mainly though I figured it like this: if someone wants to upp the number of StreamThreads that probably means they have more capacity and would want to increase the number of async threads with it, to keep the async threads per StreamThread a constant. That was just my reasoning at the time -- I can see good arguments for doing it as a "static" config. WDYT @agavra  (cc @rodesai )

Reply:
> then the actual number of threads may not match
I imagined it would be a shared pool across all threads so you can easily reason about the concurrency to the remote system

Reply:
This is a tricky one. 

I think we may want to adjust based on feedback, but if one of the concerns is the managing capacity of the external system, then having it capped statically makes sense. Basically the question is 'what is the max parallelism' that the external system can tolerate?'. Then we set the num async threads based on that. It isn't perfect though, because more Kafka Streams nodes means more threads and more parallelism anyway. 

So it probably doesn't matter that much, and we can adjust based on feedback.

Reply:
In other words, in either approach the parallelism to the external system is variable _and_ a deterministic function of the number of nodes.